src/shared/components/iframe-runtime.tsx

```diff
@@ -91,7 +92,9 @@ export const IframeRuntime: React.FC<IProps> =
       if (libraryInteractive?.localLinkedInteractiveProp && authoredState?.[libraryInteractive.localLinkedInteractiveProp]) {
         linkedInteractives = [ {id: authoredState[libraryInteractive.localLinkedInteractiveProp], label: libraryInteractive.localLinkedInteractiveProp} ];
       }
+
       phone.post("initInteractive", {
+        ...initMessage,   // for now only fullscreen sets this prop so that the needed info for cfm interactiveApi is passed
```

Comment:
When I was looking at this with @eireland, I/we had a question about the contents of the initMessage. The property at issue is that the `interactiveApiProvider` assumes that there is an `interactive` property which can be dereferenced for its `id`, presumably for the purpose of getting the `id` of the embedded interactive. When this is sent from the AP in a situation in which there is a full-screen interactive wrapping another interactive, which interactive is specified in the `interactive` prop? Is it the full-screen interactive or the embedded one? If it's the former, then shouldn't the full-screen interactive be replacing the `interactive` property with one that references the embedded interactive instead?

Reply:
In the case of the full-screen interactive, it is acting basically as a transparent proxy of the interactive it is wrapping. So I think it makes the most sense for the interactive it is wrapping to be passed the id that was originally passed by the AP (host). 

Furthermore, the host doesn't know anything about the wrapped interactive so it has no id that it could provide. This id basically represents the id of the interactive in the authoring system. 

For some tangential info about these IDs they are mentioned in a section of this doc: https://docs.google.com/document/d/1vhRfcrxPRMtJRkHDqYbDPXKPDCqZV5JjjKoSmqB5zMg/edit#heading=h.3ig3ztmzycvi
Note however, that since that section was written, the format of the ids has been updated as indicated in a following section: https://docs.google.com/document/d/1vhRfcrxPRMtJRkHDqYbDPXKPDCqZV5JjjKoSmqB5zMg/edit#heading=h.m617becjaecd

Coming back to the original problem, when the interactive being wrapped is being used in the carousel or side-by-side  which support wrapping multiple interactives the id is a problem. With Doug's change the id is still not set when this is used by the carousel or side-by-side.  It isn't clear what to should be set to in these cases. It seems like it should be unique for each of the interactives. However, the current way the logging and reporting services work it is probably best that it is the same as the top level interactive. That is because these systems will only know about this top level id, so when they are trying to group data together they will only find the data if it has this top level id.
Another potential issue is how attachments work. I believe the attachment code on the host and in the interactive depends on this id being unique. This way attachments saved by one interactive do not conflict with attachments with the same name saved by another interactive. 

So the summary is that this id issue is problematic for these nested interactives. Hopefully we'll run into this when working on the reporting of the side-by-side interactive which is scheduled next week. And we can figure out how bad the problem is then.